server/src/main/java/org/opensearch/cluster/routing/allocation/decider/ThrottlingAllocationDecider.java

```diff
@@ -150,36 +165,73 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
             // Peer recovery
             assert initializingShard(shardRouting, node.nodeId()).recoverySource().getType() == RecoverySource.Type.PEER;
 
-            // Allocating a shard to this node will increase the incoming recoveries
-            int currentInRecoveries = allocation.routingNodes().getIncomingRecoveries(node.nodeId());
-            if (currentInRecoveries >= concurrentIncomingRecoveries) {
+            if(shardRouting.unassignedReasonIndexCreated()) {
+                return allocateInitialReplicas(shardRouting, node, allocation);
+            } else {
+                return allocateNonIntialReplicas(shardRouting, node, allocation);
+            }
+        }
+    }
+
+    private Decision allocateInitialReplicas(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
+        int currentInRecoveries = allocation.routingNodes().getInitialIncomingRecoveries(node.nodeId());
+        assert shardRouting.unassignedReasonIndexCreated() && !shardRouting.primary();
+
+        return allocateReplicas(shardRouting, allocation, currentInRecoveries, replicasInitialRecoveries,
+            (x,y) -> getInitialPrimaryNodeOutgoingRecoveries(x,y), replicasInitialRecoveries,
+            String.format(Locale.ROOT, "[%s=%d]", CLUSTER_ROUTING_ALLOCATION_NODE_INITIAL_REPLICAS_RECOVERIES_SETTING.getKey(),
+                replicasInitialRecoveries),
+            String.format(Locale.ROOT, "[%s=%d]", CLUSTER_ROUTING_ALLOCATION_NODE_INITIAL_REPLICAS_RECOVERIES_SETTING.getKey(),
+                replicasInitialRecoveries));
+    }
+
+    private Decision allocateNonIntialReplicas(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
```

Comment:
1. typo
2. assert on non-initial replica

Reply:
For my understanding, is this all but initial replicas?

Reply:
1. typo- will fix.
2. private method and already if statement above. Do we really need to assert on something that was part of if block?
3. Yes, this is all but initial replicas.

Reply:
private method can still be called from another method within the same class, hence assertion?

Reply:
okay :)